src/GafferSceneUI/ContextAlgo.cpp

```diff
@@ -50,6 +51,43 @@ namespace
 InternedString g_expandedPathsName( "ui:scene:expandedPaths" );
 InternedString g_selectedPathsName( "ui:scene:selectedPaths" );
 
+bool expandWalk( const ScenePlug::ScenePath &path, const ScenePlug *scene, size_t depth, PathMatcher &expanded, PathMatcher &leafPaths )
+{
+	bool result = false;
+
+	ConstInternedStringVectorDataPtr childNamesData = scene->childNames( path );
+	const std::vector<InternedString> &childNames = childNamesData->readable();
+
+	if( childNames.size() )
+	{
+		result |= expanded.addPath( path );
```

Comment:
Right, yep, that makes sense. I've made a suggestion for a comment to add to the ContextAlgo header that I hope goes some way to clarifying this. Might need a few suggestions from you to make it clearer...